src/ansys/geometry/core/designer/body.py

```diff
@@ -1162,6 +1198,40 @@ def tessellate(  # noqa: D102
         else:
             return comp
 
+    @protect_grpc
+    def shell_body(self, offset: Real) -> bool:  # noqa: D102
+        check_type(offset, Real)
```

Comment:
You could use the decorator here for ``check_input_types``. All your type hints are properly imported.